### PR TITLE
Add asn1c/0.9.28

### PR DESCRIPTION
**Pull request by theirix**
Specify library name and version:  **asn1c/0.9.28**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/asn1c/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
```
Comment (uilianries) [inline recipes/asn1c/all/conanfile.py:2]:
```suggestion
from conan import ConanFile
from conan.tools.files import get, rmdir, chdir
from conans import AutoToolsBuildEnvironment, tools
from conan.errors import ConanInvalidConfiguration
```

File: recipes/asn1c/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import os
+
+required_conan_version = ">=1.33.0"
```
Comment (uilianries) [inline recipes/asn1c/all/conanfile.py:5]:
```suggestion
required_conan_version = ">=1.47.0"
```

File: recipes/asn1c/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    def validate(self):
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("Visual Studio is not supported")
+
+    def package_id(self):
+        del self.info.settings.compiler
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
```
Comment (uilianries) [inline recipes/asn1c/all/conanfile.py:45]:
```suggestion
        get(self, **self.conan_data["sources"][self.version],
```

File: recipes/asn1c/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        conf_args = [
+                "--datarootdir={}".format(tools.unix_path(self._datarootdir)),
+        ]
+        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def build(self):
+        with tools.chdir(self._source_subfolder):
```
Comment (uilianries) [inline recipes/asn1c/all/conanfile.py:59]:
```suggestion
        with chdir(self, self._source_subfolder):
```

File: recipes/asn1c/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            self.run("{} -fiv".format(tools.get_env("AUTORECONF")), win_bash=tools.os_info.is_windows)
+        autotools = self._configure_autotools()
+        autotools.make()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        autotools = self._configure_autotools()
+        autotools.install()
+        tools.rmdir(os.path.join(self.package_folder, "res", "doc"))
+        tools.rmdir(os.path.join(self.package_folder, "res", "man"))
```
Comment (uilianries) [inline recipes/asn1c/all/conanfile.py:69]:
```suggestion
        rmdir(self, os.path.join(self.package_folder, "res", "doc"))
        rmdir(self, os.path.join(self.package_folder, "res", "man"))
```

File: recipes/asn1c/all/test_package/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+from conans import ConanFile, CMake, tools
```
Comment (uilianries) [inline recipes/asn1c/all/test_package/conanfile.py:1]:
```suggestion
from conans import CMake
from conan import ConanFile
from conan.tools.build import cross_building
```

File: recipes/asn1c/all/test_package/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        self.build_requires(str(self.requires["asn1c"]))
+
+    def build(self):
+        shutil.copy(os.path.join(self.source_folder, "MyModule.asn1"), self.build_folder)
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self):
```
Comment (uilianries) [inline recipes/asn1c/all/test_package/conanfile.py:20]:
```suggestion
        if not cross_building(self):
```

**Review by uilianries — COMMENTED** [review]
Conan Center Index is moving toward to Conan 2.0

Some linters  are checking if new PRs are minimally compatible with Conan 2.0. As some methods will be removed in the future, the linter asks to update them.

**Comment by theirix** [comment]
@uilianries , thanks for suggestions! What a coincidence, I have already prepared almost the same changes :-)

Anyway I was confused by a few problems.

1. How to replace a very popular `os_info`? I do not see equivalents for [`tools.os_info` and others](https://github.com/conan-io/conan/blob/develop/conans/client/tools/oss.py) in `conan` package. 

2. Is it right that for 2.0-ready recipes we have not have any `conans` imports? Docs [stated that](https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#python-import-statements). If so, should we place feature requests to the Conan?

3. Conan-center's pylint-based [v2 linter](https://github.com/conan-io/conan-center-index/blob/master/docs/v2_linter.md) is amazing. How can I run it locally?

**Comment by uilianries** [comment]
@theirix 

> How to replace a very popular os_info? I do not see equivalents for [tools.os_info and others](https://github.com/conan-io/conan/blob/develop/conans/client/tools/oss.py?rgh-link-date=2022-08-03T13%3A48%3A05Z) in conan package.

It's still not available (ported). I suggest you opening an issue to Conan client, because they are removing many things and if you explain your case, there is a chance of porting it.

> Is it right that for 2.0-ready recipes we have not have any conans imports? Docs [stated that](https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#python-import-statements). If so, should we place feature requests to the Conan?

True, some of them being migrated. For instance, `minimum_cpp_required` requires Conan 1.50, which is not running Conan Center yet.  Placing feature request to keep tools which you judge as important is the best move, as more people can have same situation.

> Conan-center's pylint-based [v2 linter](https://github.com/conan-io/conan-center-index/blob/master/docs/v2_linter.md?rgh-link-date=2022-08-03T13%3A48%3A05Z) is amazing. How can I run it locally?

Excellent question! Indeed, it deserves a documentation. In short term, you need to copy the linter folder locally (which you already have with your git clone to conan-center-index), then you need to run those steps listed on Github Actions: https://github.com/conan-io/conan-center-index/blob/master/.github/workflows/linter-conan-v2.yml#L82. Of course, don't forget to install pylint first.

For instance: 

    $ pylint --rcfile=linter/pylintrc_recipe recipes/boost/all/conanfile.py

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**